src/main/groovy/io/seqera/wave/service/blob/impl/KubeTransferStrategy.groovy

```diff
@@ -65,4 +83,18 @@ class KubeTransferStrategy implements TransferStrategy {
                 .putUnencodedChars(info.creationTime.toString())
                 .hash()
     }
+
+    void cleanup(String podName) {
+        try {
+            def pod = k8sService.getPod(podName)
+            if(pod.status.phase == 'Succeeded') {
+                k8sService.deletePod(podName)
+            }else if(pod.status.phase == 'Running'){
```

Comment:
I think also this logic can be make part of `deletePodWhenReachStatus`

Reply:
correct, this is not needed `deletePodWhenReachStatus` can handle all scenarios

Reply:
done